Add module to interact with JSON datatrans API

TatianaPan opened the pull request.

Zinggi commented on lib/datatrans/config.rb:
```diff
@@ -67,5 +77,13 @@ def web_transaction(*args)
+      JSON::Transaction.new(self, *args)
+    end
+
+    def start_json_transaction(*args)
```
The other *_transaction() calls don't have an equivalent for this. Do we need it? Couldn't the caller just call `.transaction_path` on the `JSON::Transaction` themselves?

TatianaPan replied:
Good point, will change it

Zinggi commented on spec/json/authorize_spec.rb:
```diff
@@ -0,0 +1,79 @@
+describe Datatrans::JSON::Transaction::Authorize do
+  before do
+    @successful_response = {
+      "transactionId" => "230208152517866296"
```
Same comment for the `spec/json/status_spec.rb`

TatianaPan replied:
I actually took data from API examples, so I guess they are invented. bUt it's ok, I will fake them explicitly :)

Zinggi reviewed (changes requested):
Looking pretty good :+1: 
I was expecting at least one breaking change, but looks like this can easily be released as a minor update.

TatianaPan commented on lib/datatrans/json/transaction.rb:
```diff
@@ -0,0 +1,32 @@
+  end
+end
+
+require 'datatrans/json/transaction/authorize'
```
Found partial answer [here](https://stackoverflow.com/a/56739372/12769735). I followed the pattern from other parts of the gem. I guess what happens here is: 
- we define `Datatrans::JSON::Transaction` which is used then in `datatrans/json/transaction/response`, hence we need class definition before loading its subclasses

TatianaPan replied:
Otherwise I got an error `uninitialized constant Datatrans::JSON` when loading `./lib/datatrans/json/transaction/response.rb`

Zinggi replied:
Ok, I guess this is fine then

TatianaPan commented:
@Zinggi I addressed your comments and updated the Readme. We may or may not merge this now. Or we can merge, but release a bit later (when Andy is back).

Zinggi reviewed (approved):
Very nice, great job :+1:
